### ISSUE TITLE
SA1009 should not fire when it's the first character on a line

### ISSUE THREAD

**Issue by hvanbakel** [DotNetAnalyzers/StyleCopAnalyzers#1206]
This is actually two fold, my code was this:
```CSharp
            if (
                symbol == null || // Symbol can be null here if Cast is to a generic parameter for example
                symbol.Equals(declaredType) // If the cast/as/is is to the type in which this statement is contained
                )
```

After applying the fix (on line 4) it became this
```CSharp
            if (
                symbol == null || // Symbol can be null here if Cast is to a generic parameter for example
                symbol.Equals(declaredType) // If the cast/as/is is to the type in which this statement is contained
)
```
Which still violates the rule.

The violation after the fix is a bug. 

For what it's worth, I reformatted it to this to get rid of the violation.
```CSharp
            if (// Symbol can be null here if Cast is to a generic parameter for example
                symbol == null ||

                // If the cast/as/is is to the type in which this statement is contained
                symbol.Equals(declaredType))
```

**Comment by sharwell** [comment]
My understanding of SA1009 and SA1111 (ClosingParenthesisMustBeOnLineOfLastParameter) is this code should report a violation. Since SA1111 only applies to method definitions and calls, SA1009 would be the warning to report here.

I would have expected the code fix to produce this output:

```CSharp
            if (
                symbol == null || // Symbol can be null here if Cast is to a generic parameter for example
                symbol.Equals(declaredType)) // If the cast/as/is is to the type in which this statement is contained
```

**Comment by hvanbakel** [comment]
Exactly, even though this will introduce a new violation on the comment being not within a block (if I recall correctly).
